frontend/packages/dev-console/src/components/topology/TopologyApplicationResources.tsx

```diff
@@ -20,7 +20,12 @@ const TopologyApplicationResources: React.FC<TopologyApplicationResourcesProps>
   const resourcesData = {};
   _.forEach(resources, (res) => {
     const a = getTopologyResourceObject(res);
-    resourcesData[a.kind] = [...(resourcesData[a.kind] ? resourcesData[a.kind] : []), a];
+    let key = a.kind;
+    const isKnativeResource = _.get(res, 'data.isKnativeResource', false);
+    if (isKnativeResource) {
+      key = referenceForModel(modelFor(referenceFor(a)));
```

Comment:
@invincibleJai cant we simply do this `referenceForModel(modelFor(a.kind))` ?

Reply:
nope it's a CRD based resource so had to call `referenceFor` as for example kind is same `Service` for k8's service and knative service

Reply:
Oh okay. Got it

Reply:
@divyanshiGupta thanks for pointing it, was discussing with @karthikjeeyar it can be handled just with `referenceFor(a)` as I din't needed whole model back